samples/brokers/onvif-video-broker/Akri.cs

```diff
@@ -102,6 +128,16 @@ private static string GetStreamingUri(String media_url, String profile_token)
 			// randomly choose first profile
 			var streaming_uri = streaming_uri_list.First();
 			Console.WriteLine($"[Akri] ONVIF streaming uri {streaming_uri}");
+
+			const string rtspPrefix = "rtsp://";
+			if (streaming_uri.StartsWith(rtspPrefix)) {
+				if (!String.IsNullOrEmpty(usernameToken.Username)) {
+					var password = usernameToken.Password ?? "";
```

Comment:
curious: is having a valid username without a password a valid scenario?

Reply:
maybe if this should be an error: `(!String.IsNullOrEmpty(usernameToken.Username) && String.IsNullOrEmpty(usernameToken.Password))`

Reply:
password can be optional, empty string is used if not specified.